docs/release/versioning-policy.md

```diff
@@ -0,0 +1,23 @@
+# Kubeflow Pipelines Versioning Policy
+
+The Kubeflow Pipelines version format follows [the semantic versioning](https://semver.org/).
+The Kubeflow Pipelines versions are in the format of `X.Y.Z`, where `X` is the major version,
+`Y` is the minor version, and `Z` is the patch version.
+
+As a guideline, we increment the:
+* MAJOR version when we make breaking changes on [stable features](./feature-status.md#stable-features),
+* MINOR version when we add functionality in a backwards compatible manner, and
+* PATCH version when we make backwards compatible bug fixes.
+
+In reality, we are not following the guideline strictly, e.g. PATCH versions might have backward compatible new functionalities that is low-risk.
```

Comment:
Suggested revision:

In reality, we are not following the guideline strictly. For example, a PATCH version might include a backward compatible new feature that is considered to be low-risk.